### PR TITLE
[EC-548] Member Details Group Tab

### PR DESCRIPTION
**Pull request by shane-melton**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Implement the Groups tab in the member details dialog. 

There is a required Server PR https://github.com/bitwarden/server/pull/2508

## Code changes

- **organizations/members/components/member-dialog/member-dialog.component:** Add group tab implementation and cleanup duplicated data fetching
- **organizations/core/services/user-admin.service.ts:** Add group ids to the view
- **organization-user/requests/organization-user-invite.request.ts:** Add group ids to the view

## Screenshots

image

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts [hunk 1]
```diff
@@ -79,11 +76,13 @@ export class MemberDialogComponent implements OnInit, OnDestroy {
 
   protected organization: Organization;
   protected accessItems: AccessItemView[] = [];
```
Comment (coroiu) [inline apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts:78]: suggestion(non-blocking): we could probably rename this to `collectionAccessItems` just for consistency

File: apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts [hunk 2]
```diff
@@ -233,30 +204,28 @@ export class MemberDialogComponent implements OnInit, OnDestroy {
...
+          groups.map<AccessItemView>((g) => mapGroupToAccessItemView(g))
+        );
+
         if (this.params.organizationUserId) {
           if (!userDetails) {
             throw new Error("Could not find user to edit.");
           }
 
           const accessSelections = mapToAccessSelections(userDetails);
+          const groupAccessSelections = mapToGroupAccessSelections(userGroups);
```
Comment (coroiu) [inline apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts:217]: question: what about fetching userGroups together with user details?

**Review by coroiu — COMMENTED** [review]
Nice and small, good work! The only question I think we should answer is around data fetching

File: apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts (hunk 2 above)
Comment (shane-melton) [reply apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts]: Oh yep, updated to use the same API call!

File: apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts (hunk 1 above)
Comment (shane-melton) [reply apps/web/src/app/organizations/members/components/member-dialog/member-dialog.component.ts]: Done!

**Review by coroiu — APPROVED** [review]

**Pull request merged**